### ISSUE TITLE
False-negative `unsubscriptable-object` with `__class_getitem__` and Python 3.6

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#5358]
### Current problem

`__class_getitem__` was specified in PEP 560 and implement in Python 3.7
https://www.python.org/dev/peps/pep-0560/#class-getitem

If `py-version` is set to `< 3.7`, it would be nice if we could emit errors in the example below.

```py
class Parent:
    def __class_getitem__(cls, item):
        return cls

Parent[int]  # should raise 'unsubscriptable-object'

class Child(Parent[int]):  # should raise 'inherit-non-class'
    ...
```

### Desired solution

Modify `supports_getitem` helper to respect `py-version` setting.
https://github.com/PyCQA/pylint/blob/d29bcf68393b6a1654aa6cd3d5351ed63edb42ad/pylint/checkers/utils.py#L1203-L1209

### Additional context

Subscription via PEP 585 in annotations (with `from __future__ import annotations`) could also be update. (Requires Python 3.7)

It could make sense to modify the error message in these cases to mention that it's unsubscriptable in "some" supported Python versions. (if possible)

### For reference
* https://github.com/PyCQA/astroid/pull/1266#discussion_r753820025
* [inherit_non_class.py#L94-L97](https://github.com/cdce8p/pylint/blob/d29bcf68393b6a1654aa6cd3d5351ed63edb42ad/tests/functional/i/inherit_non_class.py#L94-L97)

**Comment by Pierre-Sassoulas** [comment]
Closing as we don't support python 3.6 anymore.
